src/sleep/sleep.c

```diff
@@ -588,6 +588,23 @@ static int parse_argv(int argc, char *argv[]) {
         return 1 /* work to do */;
 }
 
+static int write_wakeup_count(void) {
+        static const char wakeup_count_path[] = "/sys/power/wakeup_count";
+        _cleanup_free_ char *buf = NULL;
+        size_t bufsize;
+        int r;
+
+        r = read_full_virtual_file(wakeup_count_path, &buf, &bufsize);
+        if (r < 0)
+                return log_debug_errno(r, "Unable to read %s: %m", wakeup_count_path);
+
+        r = write_string_file(wakeup_count_path, buf, WRITE_STRING_FILE_DISABLE_BUFFER);
```

Comment:
so you read a binary (i.e. sized) blob, but you write a nul terminated string. This actually will likely work, since read_full_virtual_file() will silently NUL terminate for safety reasons even if you specify a size. But it's still a bit asymmetric. please pass NULL as third param to read_full_virtual_file() which tells it that you intend to read this as a NUL terminated string, not as a binary blob, and it will take that into account (for example, look for embedded NUL bytes, and refuse them). it also shortens the code...

Reply:
Thanks, adjusted.